src/docs/05-go-client/18-workflow-replay-shadowing.md

```diff
@@ -0,0 +1,176 @@
+---
+layout: default
+title: Workflow Replay and Shadowing
+permalink: /docs/go-client/workflow-replay-shadowing
+---
+
+# Workflow Replay and Shadowing
+
+In the Versioning section, we mentioned that incompatible changes to workflow definition code could cause non-deterministic issues when processing workflow tasks if versioning is not done correctly. However, it's may be hard for you to tell if a particular change is incompatible or not and whether versioning logic is needed. To help you identify incompatible changes and catch them before production traffic is impacted, we implemented Workflow Replayer and Workflow Shadower.
+
+## Workflow Replayer
+
+Workflow Replayer is a testing component for replaying existing workflow histories against a workflow definition. The replaying logic is the same as the one used for processing workflow tasks, so if there's any incompatible changes in the workflow definition, the replay test will fail. 
+
+### Write a Replay Test
+
+#### Step 1: Create workflow replayer
+
+Create a workflow Replayer by:
+
+```go
+replayer := worker.NewWorkflowReplayer()
+```
+or if custom data converter, context propagator, interceptor, etc. is used in your workflow:
+
+```go
+options := worker.ReplayOptions{
+  DataConverter: myDataConverter,
+  ContextPropagators: []workflow.ContextPropagator{
+    myContextPropagator,
+  },
+  WorkflowInterceptorChainFactories: []interceptors.WorkflowInterceptorFactory{
+    myInterceptorFactory,
+  },
+  Tracer: myTracer,
+}
+replayer := worker.NewWorkflowReplayWithOptions(options)
+```
+
+#### Step 2: Register workflow definition
+
+Next, register your workflow definitions as your normally do. Make sure workflows are registered the same way as they were when running and generating histories; otherwise the replay will not able to find the corresponding definition.
```

Comment:
> as your normally do
* you

> otherwise the replay will not able to find the corresponding definition.
* otherwise the replay will not **be** able to find the corresponding definition.